pontoon/base/models/comment.py

```diff
@@ -28,10 +28,15 @@ def __str__(self):
 
     def serialize(self):
         locale = self.locale or self.translation.locale
+        project = (
+            self.translation.entity.resource.project
+            if self.translation
+            else self.entity.resource.project
+        )
```

Comment:
This is not very pretty and not very fast. Similarly, the path to get to `project` in [pontoon/base/views.py](https://github.com/mozilla/pontoon/pull/3422/files#diff-f4cdb2e09580f4e478d1804c3b86f26eed7e43f1f2fb859e83d326074c52d560), while a bit prettier, is still slow.

There are ways to get to the project object in a more scalable way, e.g. by passing the project from frontend all the way to here or by passing `contact = entity.resource.project.contact` instead.

My personal preference would be to just use `user.groups.filter(name="project_managers").exists()` in `user_status()`, which would show all PMs, regardless of the project, as PMs. 

I think that's even better, because when PMs of other projects add translations or comments, they are effectively acting PMs (e.g. because they step in temporarily or because they have managed the project in the past).

Reply:
> My personal preference would be to just use user.groups.filter(name="project_managers").exists() in user_status(), which would show all PMs, regardless of the project, as PMs.

That kind of defies the main point of having the PM banner. 

I want the PM banner to show up only within the project for which that user is set as PM, not in all of them.

An admin that is also a PM should show up as an Admin. A person that is not an Admin, but is set as a PM for a specific project, shouldn't be displayed as a PM for any other projects (sure, they're unlikely to do anything outside of their project, but we should still avoid presenting them as something they're not).

Regarding this being slow: can we live with it, considering it's only used when there are actually comments (so, not in the majority of cases)? We have 12k comments for almost 8M translations, and I bet most comments are in group of at least 2 (so, ballpark of 4–5k translations affected? I'm sure actual data can be extracted if we think it's important).

Reply:
> There are ways to get to the project object in a more scalable way, e.g. by passing the project from frontend all the way to here or by passing contact = entity.resource.project.contact instead.

I'd be happy to have someone else pick up this PR and pursue this path, because I'm not sure that's something I can do with my current understanding of the codebase 😉

Reply:
(investigating the `passing contact = entity.resource.project.contact instead` in the meantime)

Reply:
I'll push updates to your branch directly.

Performance degrades for every string that has translations. The PR makes 3 additional DB queries in [pontoon/base/views.py](https://github.com/mozilla/pontoon/pull/3422/files#diff-f4cdb2e09580f4e478d1804c3b86f26eed7e43f1f2fb859e83d326074c52d560) for each translation loaded in the History panel if translation **doesn't** have any comments. 4 DB queries are made here **on top of that** for translations with comments, for a total of 7 additional DB queries. We should avoid that.